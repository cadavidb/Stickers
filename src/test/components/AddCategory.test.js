import React from 'react'
import '@testing-library/jest-dom';
const { AddCategory } = require("../../components/AddCategory")
import {shallow} from 'enzyme'




describe('pruebas sobre <AddCategory>', () => {
    const setCate=jest.fn();
    let wrapper;
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper=shallow(<AddCategory setCategories={setCate}/>)
    })
    
    test('debe mostrar el componente correctamente', () => {
       
        expect(wrapper).toMatchSnapshot()
    })


    test('debe de cambiar la caja de texto', () => {

        const input=wrapper.find('input');
        const value="holaa"
        input.simulate('change',{target:{value}})
        expect(wrapper.find('p').text().trim()).toBe(value)
    
    })


    test('No debe postear informacion con submit', () => {

        wrapper.find('form').simulate('submit',{preventDefault(){}})
        expect(setCate).not.toHaveBeenCalled();
    })
    

    //task

    test('must call function setcategories and clear the text box', () => {
        //simular el input change
        const input=wrapper.find('input');
        const value="simular input pruebas unitarias"
        input.simulate('change',{target:{value}})
        
        //simular el submit 
        wrapper.find('form').simulate('submit',{preventDefault(){}})
        
        // //set categories se debe haber llamado
         expect(setCate).toHaveBeenCalled();
         expect(setCate).toHaveBeenCalledTimes(1);
         expect(setCate).toHaveBeenCalledWith(expect.any(Function));
         
         
       //el valor del input debe estar '';
       expect(input.prop('value')).toBe('')



    })
    
    
    
    
})
