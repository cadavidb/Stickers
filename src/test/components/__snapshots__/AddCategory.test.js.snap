// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre <AddCategory> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
       
       
    </p>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="animate__animated animate__shakeY"
      onClick={[Function]}
    >
      Buscar
    </button>
  </form>
</Fragment>
`;
